beep/tests/test_protocol.py

```diff
@@ -415,6 +414,89 @@ def test_prediag_with_waveform(self):
+                    template_fullpath = os.path.join(PROCEDURE_TEMPLATE_DIR, template)
+                    protocol = Procedure.generate_procedure_drivingv1(index,
+                                                                      protocol_params,
+                                                                      waveform_name,
+                                                                      template=template_fullpath)
+                    protocol.generate_procedure_diagcyclev3(
+                        protocol_params["capacity_nominal"], diagnostic_params
+                    )
+                    filename = "{}.000".format(filename_prefix)
+                    filename = os.path.join(output_directory, "procedures", filename)
+                    waveform_names.append(os.path.split(waveform_name)[-1])
+
+                if not os.path.isfile(filename):
+                    protocol.to_file(filename)
+                    new_files.append(filename)
+                    names.append(filename_prefix + "_")
+
+                self.assertEqual(protocol.get("MaccorTestProcedure.ProcSteps.TestStep.32.StepType"), 'FastWave')
+                self.assertEqual(protocol.get("MaccorTestProcedure.ProcSteps.TestStep.64.StepType"), 'FastWave')
+                wave_value = os.path.split(waveform_name)[-1].split(".")[0]
+                self.assertEqual(protocol.get("MaccorTestProcedure.ProcSteps.TestStep.32.StepValue"),
+                                 wave_value)
+                self.assertEqual(protocol.get("MaccorTestProcedure.ProcSteps.TestStep.64.StepValue"),
+                                 wave_value)
+
+            self.assertEqual(len(os.listdir(os.path.join(output_directory, "procedures"))), 36)
+            self.assertEqual(len(os.listdir(os.path.join(output_directory, "mwf_files"))), 18)
+
+        test_names = ['US06_x4_24W.MWF', 'LA4_x4_10W.MWF',
+                      'US06_x4_32W.MWF', 'LA4_x4_14W.MWF',
+                      'US06_x4_40W.MWF', 'LA4_x4_18W.MWF',
+                      'US06_x8_24W.MWF', 'LA4_x8_10W.MWF',
+                      'US06_x8_32W.MWF', 'LA4_x8_14W.MWF',
+                      'US06_x8_40W.MWF', 'LA4_x8_18W.MWF',
+                      'US06_x12_24W.MWF', 'LA4_x12_10W.MWF',
+                      'US06_x12_32W.MWF', 'LA4_x12_14W.MWF',
+                      'US06_x12_40W.MWF', 'LA4_x12_18W.MWF',
+                      'US06_x4_24W.MWF', 'LA4_x4_10W.MWF',
+                      'US06_x4_32W.MWF', 'LA4_x4_14W.MWF',
+                      'US06_x4_40W.MWF', 'LA4_x4_18W.MWF',
+                      'US06_x8_24W.MWF', 'LA4_x8_10W.MWF',
+                      'US06_x8_32W.MWF', 'LA4_x8_14W.MWF',
+                      'US06_x8_40W.MWF', 'LA4_x8_18W.MWF',
+                      'US06_x12_24W.MWF', 'LA4_x12_10W.MWF',
+                      'US06_x12_32W.MWF', 'LA4_x12_14W.MWF',
+                      'US06_x12_40W.MWF', 'LA4_x12_18W.MWF']
```

Comment:
This is great! Can you add an assert statement to ensure that the max(abs(power)) in the MWF file does not exceed what we expect, just to be safe? Maybe for the 40 W ones?